web/src/lib/utils/context.ts

```diff
@@ -0,0 +1,8 @@
+import { getContext, setContext } from 'svelte';
+
+export function createContext<T>(key: string | symbol = Symbol()) {
```

Comment:
Can you help me understand why we need this? I read the docs but didn't understand them very well. https://svelte.dev/tutorial/context-api

Reply:
Other solutions are possible, but using context seemed best suited. You can use props, but then you have to pass the map to every component with something like:

```svelte
<Map bind:map={map}>
  <TileLayer {map} />
  <Marker {map} />
</Map>
```

That isn't needed with contexts. You set the `map` context once in the `Map` component and all children can access it. This is better manageable, especially when adding more components or even adding more contexts.

You can also use a store, but if you have two maps on one page, they would both interact with the same store leading to all sorts of issues. This doesn't happen with contexts, because they are limited to a single instance of a component.